models/sales/intermediate/int_hubspot__deals_enhanced.sql

```diff
@@ -5,6 +5,20 @@ with deals as (
     select *
     from {{ var('deal') }}
 
+), merged_deals as (
+
+    select *
+    from {{ var('merged_deal')}}
+
+), aggregate_merged_deals as (
+
+    select
+        deal_id,
+        {{ fivetran_utils.array_agg("merged_deal_id") }} as merged_deal_ids
+    
+    from {{ var('merged_deal')}}
+    group by 1
+
```

Comment:
Why are we doing both of these ctes? Wouldn't just the second suffice, or just select from the `merged_deals` cte as opposed to the variable in the second cte?

Reply:
the 2nd CTE `aggregate_merged_deals` is made because the `merged_deals` table can have duplicative deal_ids, but I want to get it to a unique deal_id grain.  But that's a good point, I switched from selecting from the variable to just the `merged_deals` cte